virtualizarr/zarr.py

```diff
@@ -133,28 +112,12 @@ def to_kerchunk_json(self) -> str:
 
     def replace(
         self,
-        chunks: Optional[tuple[int, ...]] = None,
-        compressor: Optional[dict] = None,
-        dtype: Optional[np.dtype] = None,
-        fill_value: Optional[float] = None,  # float or int?
-        filters: Optional[list[dict]] = None,  # type: ignore[valid-type]
-        order: Optional[Literal["C"] | Literal["F"]] = None,
-        shape: Optional[tuple[int, ...]] = None,
-        zarr_format: Optional[Literal[2] | Literal[3]] = None,
+        **kwargs: Any,
```

Comment:
Do we want the list of fields that can be replaced here in the signature?

Reply:
As opposed to letting users call [dataclasses.replace](https://docs.python.org/3/library/dataclasses.html#dataclasses.replace)? I think `ZArray` shouldn't have fields that aren't replaceable, are there any fields like that?

Reply:
I think if `dataclasses.replace` exists already then explicitly writing the list of fields isn't a good enough reason to overwrite the default method.

Reply:
Oh no wait `dataclasses.replace` is a module-level function, not a method on the dataclass type. In that case I change my mind - we should list out the arguments and call `dataclasses.replace` from within the `ZArray.replace` method.

Reply:
> I think ZArray shouldn't have fields that aren't replaceable, are there any fields like that?

Just to be clear, the behavior here (and on `main`) is to return a new instance of `ZArray` with the same values as the old instance, but new values specified in the keyword arguments.

> we should list out the arguments

Sounds good. I think we'll be able to do that (we'll need to remember to keep the two signatures in step though).